service/storage_proxy.cc

```diff
@@ -4148,25 +4153,35 @@ storage_proxy::query_singular(lw_shared_ptr<query::read_command> cmd,
         };
 
         if (exec.size() == 1) [[likely]] {
-            result = (co_await exec[0].first->execute(timeout)).value(); // TODO: This rethrows the exception in case of failed result
-            handle_completion(exec[0]);
+            result = co_await exec[0].first->execute(timeout);
+            if (result) {
+                handle_completion(exec[0]);
+            }
```

Comment:
else? the original code propagates the error.

I see if (!result) below, it's somewhat odd to have an if/else split into two ifs. Suggest to add a comment that explains the failure case here.

Reply:
Added comments to this function which explain that the failure is handled later (in the `if (!result) {}` block).